Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -134,4 +290,51 @@ extension PreplannedMapArea: PreplannedMapAreaProtocol {
     var description: String {
         portalItem.description
     }
+    
+    var id: PortalItem.ID? {
+        portalItem.id
+    }
+}
+
+extension FileManager {
+    private static let mmpkPathExtension: String = "mmpk"
+    private static let offlineMapAreasPath: String = "OfflineMapAreas"
+    private static let packageDirectoryPath: String = "Package"
+    private static let preplannedDirectoryPath: String = "Preplanned"
+    
+    /// The path to the documents folder.
+    private var documentsDirectory: URL {
+        URL.documentsDirectory
+    }
+    
+    /// The path to the offline map areas directory within the documents directory.
+    /// `Documents/OfflineMapAreas`
+    private var offlineMapAreasDirectory: URL {
+        documentsDirectory.appending(
+            path: Self.offlineMapAreasPath,
+            directoryHint: .isDirectory
+        )
+    }
+    
+    /// The path to the web map directory for a specific portal item.
+    /// `Documents/OfflineMapAreas/<Portal Item ID>`
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    private func portalItemDirectory(forPortalItemID portalItemID: Item.ID) -> URL {
+        offlineMapAreasDirectory.appending(path: portalItemID.rawValue, directoryHint: .isDirectory)
+    }
+    
+    /// The path to the preplanned map areas directory for a specific portal item.
+    /// `Documents/OfflineMapAreas/<Portal Item ID>/Preplanned/<Preplanned Area ID>`
+    /// - Parameter portalItemID: The ID of the web map portal item.
+    func preplannedDirectory(forPortalItemID portalItemID: Item.ID, preplannedMapAreaID: Item.ID) -> URL {
```

Comment:
After this PR is merged, I can add more tests for better code coverage, like for this one.